keras_cv/layers/object_detection/non_max_suppression_test.py

```diff
@@ -23,126 +23,136 @@ def test_return_shapes(self):
         images = tf.ones((2, 256, 512, 1))
-        boxes = layer(predictions, images)
-        self.assertEqual(boxes.shape, [2, None, 6])
-
-    def test_different_channels(self):
-        layer = NonMaxSuppression(classes=4, bounding_box_format="xyWH")
-        images = tf.ones((3, 480, 480, 5))
-
-        boxes = tf.cast(tf.random.uniform((3, 5, 4), 0, 480, tf.int32), tf.float32)
-        classes = tf.cast(tf.random.uniform((3, 5, 1), 0, 4, tf.int32), tf.float32)
-        scores = tf.random.uniform((3, 5, 1), 0, 1, tf.float32)
-
-        predictions = tf.concat([boxes, classes, scores], axis=-1)
-
-        boxes = layer(predictions, images)
-        self.assertEqual(boxes.shape, [3, None, 6])
-
-    def test_in_a_model(self):
-        input1 = tf.keras.layers.Input([5, 6])
-        input2 = tf.keras.layers.Input([480, 480, 3])
-        layer = NonMaxSuppression(classes=4, bounding_box_format="xyWH")
-        outputs = layer(input1, input2)
-
-        model = tf.keras.models.Model(inputs=[input1, input2], outputs=outputs)
-
-        images = tf.ones((3, 480, 480, 3))
-
-        boxes = tf.cast(tf.random.uniform((3, 5, 4), 0, 480, tf.int32), tf.float32)
-        classes = tf.cast(tf.random.uniform((3, 5, 1), 0, 4, tf.int32), tf.float32)
-        scores = tf.random.uniform((3, 5, 1), 0, 1, tf.float32)
-
-        predictions = tf.concat([boxes, classes, scores], axis=-1)
-
-        boxes = model([predictions, images])
-        self.assertEqual(boxes.shape, [3, None, 6])
+        outputs = layer(predictions, images)
+        self.assertEqual(outputs["boxes"].shape, [3, None, 4])
+        self.assertEqual(outputs["classes"].shape, [3, None])
+        self.assertEqual(outputs["confidence"].shape, [3, None])
 
     def test_without_images(self):
         layer = NonMaxSuppression(classes=4, bounding_box_format="xyWH")
 
         boxes = tf.cast(tf.random.uniform((3, 5, 4), 0, 480, tf.int32), tf.float32)
-        classes = tf.cast(tf.random.uniform((3, 5, 1), 0, 4, tf.int32), tf.float32)
-        scores = tf.random.uniform((3, 5, 1), 0, 1, tf.float32)
-
-        predictions = tf.concat([boxes, classes, scores], axis=-1)
+        classes = tf.cast(tf.random.uniform((3, 5), 0, 4, tf.int32), tf.float32)
+        scores = tf.random.uniform((3, 5), 0, 1, tf.float32)
+        predictions = {"boxes": boxes, "classes": classes, "confidence": scores}
 
-        boxes = layer(predictions)
-        self.assertEqual(boxes.shape, [3, None, 6])
+        outputs = layer(predictions)
+        self.assertEqual(outputs["boxes"].shape, [3, None, 4])
+        self.assertEqual(outputs["classes"].shape, [3, None])
+        self.assertEqual(outputs["confidence"].shape, [3, None])
 
     def test_ragged_output_with_differing_shapes(self):
         layer = NonMaxSuppression(8, "xywh", iou_threshold=0.1)
         images = tf.ones((2, 480, 480, 3))
 
-        predictions = tf.convert_to_tensor(
-            [
+        predictions = {
+            "boxes": tf.convert_to_tensor(
```

Comment:
This also looks like it got sorta nuked by the autoformatter

Reply:
Yeah..I don't understand why it didn't format this file previously, but the linter chooses to run such aggressive formating in this PR

Reply:
Its the dictionary nesting - I guess the formatter handles it differently.  This sucks, let me try to fix.

Reply:
No luck, a fix would be welcome but I couldn't get it to stay nice.